google/cloud/bigquery_storage_v1/reader.py

```diff
@@ -328,10 +345,12 @@ def to_dataframe(self, dtypes=None):
         # pandas dataframe is about 2x faster. This is because pandas.concat is
         # rarely no-copy, whereas pyarrow.Table.from_batches + to_pandas is
         # usually no-copy.
-        schema_type = self._read_session._pb.WhichOneof("schema")
-
-        if schema_type == "arrow_schema":
+        try:
             record_batch = self.to_arrow()
+        except NotImplementedError:
+            record_batch = None
+
+        if record_batch is not None:
```

Comment:
Would this be clearer as:
```suggestion
        except NotImplementedError:
            pass
        else:
```